mantidproject/mantid
# Fix cans not being saved sans

**Description of work.**

- Drop confusing dict which was being abused to pack output workspaces for a data class
- Add a new field to that data class to store scaled hab. This was being output at the same time as hab data and caused the save step to fail (as we now had +1 workspace over the rest that always come in sets)
- Fix inverted logic on saving can workspaces

**To test:**
- Ensure you have the ISIS Training Data / loqdemo folder
- Open ISIS SANS GUI
- Load the .txt as the mask file
- In the same folder load the csv as the batch file
- In settings change reduction mode from all to merged
- Change the default save directory to an empty dir
- Go back to the runs table, In the bottom right ensure that file is selected, and "Save Can" is ticked
- Select the first row and process selected.
- Go to the output dir and check files with `can + hab` and `can + lab` both are output (the can files were previously missing)

Part of #29454 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
